packages/@glimmer/manager/lib/internal/index.ts

```diff
@@ -138,7 +139,18 @@ export function getInternalHelperManager(
     );
   }
 
-  const manager = getManager(HELPER_MANAGERS, definition)!;
+  let manager = getManager(HELPER_MANAGERS, definition)!;
```

Comment:
This `!` seems suspect (we still check for `undefined`, etc later on) – what is it doing for us here?

Reply:
this was pre-existing, but it's because at the end of the function `return manager` expects non-undefined. If manager ever is undefined, we could throw an error and get away from both `!`, but is that the correct behavior? happy to add if it is.

Reply:
Is typescript not able to follow the code somehow? It seems like we branched in a way that I would expect TS to be able to figure it out, since we did check for undefined.

But anyway, we can structure it like this:

```ts
let manager = ...;

if (manager) {
  return manager;
} else if (typeof definition === 'function') {
  return DEFAULT_MANAGER;
} else if (isOptional === true) { // === true seems a bit extra but whatev
  return null;
} else {
  throw new Error(...);
}
```